### DISCUSSION TITLE
Custom migrations

### DISCUSSION THREAD

**Discussion by rabet** [geldata/gel#4519] [General]
Hey guys,

can we have custom migrations with Edgedb ? I mean to execute code which modifies the db with the custom logic? 

Thanks again! 😄

**Comment by elprans** [comment]
Can you give an example of what you have in mind?

**Reply by JakubBlunar** [reply]
Basically what you need to do in that migration is transform database into new state. It is not always just adding/modifying table columns or relationships. Sometimes you need to migrate old data into new tables or do operations with old values.

basicaly you want to run custom code that will for example create temporary table, then it will fill it with data and based on these data it will do something else and at the end it will delete this temporary table. Or you will want to create table and insert it with data, other option is to modify column values.  

In type-orm you can have any code in your migration, so you can do want need to do even if it is something more complicated. 
![image](https://user-images.githubusercontent.com/6445617/195799784-f125bdae-7405-4ef1-a21d-919234745592.png)

**Reply by elprans** [reply]
EdgeDB migrations are just EdgeQL files, and you can edit them directly and place any data modifying statement in there.  We've recently made this easier with [`edgedb migration edit`](edgedb/edgedb-cli/pull/842) (to be released soon).

We're also planning to add support for migrations written using language bindings a-la the above.
